File: app/src/main/java/com/bccommittee/adapter/MemberAdapter.java
package com.bccommittee.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bccommittee.R;
import com.bccommittee.model.Member;
import com.bccommittee.util.CurrencyUtil;

import java.util.ArrayList;
import java.util.List;

public class MemberAdapter extends RecyclerView.Adapter<MemberAdapter.MemberViewHolder> {

    public interface OnMemberClickListener {
        void onMemberClick(Member member);
    }

    private final Context context;
    private final OnMemberClickListener listener;
    private List<Member> members = new ArrayList<>();

    public MemberAdapter(Context context, OnMemberClickListener listener) {
        this.context = context;
        this.listener = listener;
    }

    public void setMembers(List<Member> members) {
        this.members = new ArrayList<>(members);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public MemberViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(context).inflate(R.layout.item_member, parent, false);
        return new MemberViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull MemberViewHolder holder, int position) {
        Member m = members.get(position);
        holder.bind(m);
        holder.itemView.setOnClickListener(v -> listener.onMemberClick(m));
    }

    @Override
    public int getItemCount() { return members.size(); }

    static class MemberViewHolder extends RecyclerView.ViewHolder {
        TextView tvAvatar, tvName, tvContribution, tvJoinDate, tvLoanAmount, tvLoanLabel;
        FrameLayout frameAvatar;

        MemberViewHolder(View v) {
            super(v);
            tvAvatar       = v.findViewById(R.id.tv_avatar);
            tvName         = v.findViewById(R.id.tv_member_name);
            tvContribution = v.findViewById(R.id.tv_contribution);
            tvJoinDate     = v.findViewById(R.id.tv_join_date);
            tvLoanAmount   = v.findViewById(R.id.tv_loan_amount);
            tvLoanLabel    = v.findViewById(R.id.tv_loan_label);
            frameAvatar    = v.findViewById(R.id.frame_avatar);
        }

        void bind(Member m) {
            tvAvatar.setText(m.getAvatar() != null ? m.getAvatar() : "?");
            tvName.setText(m.getName());
            tvContribution.setText(CurrencyUtil.format(m.getContribution()) + "/mo");
            tvJoinDate.setText(m.getJoinDate() != null ? m.getJoinDate() : "");

            if (m.getAmtBorrowed() > 0) {
                tvLoanAmount.setText(CurrencyUtil.format(m.getAmtBorrowed()));
                tvLoanAmount.setTextColor(
                    itemView.getContext().getColor(R.color.color_orange));
                tvLoanLabel.setText("Loan");
            } else {
                tvLoanAmount.setText("No loan");
                tvLoanAmount.setTextColor(
                    itemView.getContext().getColor(R.color.text_muted));
                tvLoanLabel.setText("");
            }
        }
    }
}

File: app/src/main/kotlin/com/khaga/ksociety/adapter/MemberAdapter.kt
package com.khaga.ksociety.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.khaga.ksociety.R
import com.khaga.ksociety.databinding.ItemMemberBinding
import com.khaga.ksociety.model.Member
import com.khaga.ksociety.util.CurrencyUtil

class MemberAdapter(
    private val onMemberClick: (Member) -> Unit
) : ListAdapter<Member, MemberAdapter.MemberViewHolder>(MemberDiffCallback()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MemberViewHolder {
        val binding = ItemMemberBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        )
        return MemberViewHolder(binding)
    }

    override fun onBindViewHolder(holder: MemberViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    inner class MemberViewHolder(
        private val binding: ItemMemberBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        init {
            binding.root.setOnClickListener {
                onMemberClick(getItem(adapterPosition))
            }
        }

        fun bind(member: Member) {
            with(binding) {
                tvAvatar.text = member.avatar.ifEmpty { "?" }
                tvMemberName.text = member.name
                tvContribution.text = "${CurrencyUtil.format(member.contribution)}/mo"
                tvJoinDate.text = member.joinDate

                if (member.amtBorrowed > 0) {
                    tvLoanAmount.text = CurrencyUtil.format(member.amtBorrowed)
                    tvLoanAmount.setTextColor(root.context.getColor(R.color.color_orange))
                    tvLoanLabel.text = "Loan"
                } else {
                    tvLoanAmount.text = "No loan"
                    tvLoanAmount.setTextColor(root.context.getColor(R.color.text_muted))
                    tvLoanLabel.text = ""
                }
            }
        }
    }

    private class MemberDiffCallback : DiffUtil.ItemCallback<Member>() {
        override fun areItemsTheSame(old: Member, new: Member) = old.id == new.id
        override fun areContentsTheSame(old: Member, new: Member) = old == new
    }
}
